target/linux/ath79/dts/ar9330_teltonika_rut230.dts

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9330.dtsi"
+
+/ {
+	compatible = "teltonika,rut230", "qca,ar9330";
+	model = "Teltonika RUT230";
+
+	chosen {
+		bootargs = "console=ttyATH0,115200 root=31:05 rootfstype=squashfs";
+	};
+
+	aliases {
+		serial0 = &uart;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 22 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		input {
+			label = "input";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		sim {
+			label = "SIM holder";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		signal1 {
+			label = "rut230:green:signal1";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+
+		signal2 {
+			label = "rut230:green:signal2";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+		};
+
+		signal3 {
+			label = "rut230:green:signal3";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		signal4 {
+			label = "rut230:green:signal4";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
+		};
+
+		signal5 {
+			label = "rut230:green:signal5";
+			gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+		};
+
+		status_2g {
+			label = "rut230:green:status_2g";
+			gpios = <&gpio 8 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_3g {
+			label = "rut230:green:status_3g";
+			gpios = <&gpio 24 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_4g {
+			label = "rut230:green:status_4g";
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+
+		lan {
+			label = "rut230:green:lan";
+			gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		};
+
+		wan {
+			label = "rut230:green:wan";
+			gpios = <&gpio 17 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x0 0x20000>;
+				label = "u-boot";
+				read-only;
+			};
+
+			config: partition@20000 {
+				reg = <0x20000 0x10000>;
+				label = "config";
+				read-only;
+			};
+
+			art: partition@30000 {
+				reg = <0x30000 0x10000>;
+				label = "art";
+				read-only;
+			};
+
+			partition@40000 {
+				compatible = "tplink,firmware";
+				reg = <0x40000 0xf30000>;
+				label = "firmware";
+			};
+
+			partition@f70000 {
+				reg = <0xf70000 0x90000>;
+				label = "event-log";
+				read-only;
+			};
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&config 0x0>;
+
+	gmac-config {
+		device = <&gmac>;
+
+		switch-phy-addr-swap = <0>;
+		switch-phy-swap = <0>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&config 0x0>;
+	mtd-mac-address-increment = <1>;
```

Comment:
Okay, so you just need to move the `mtd-mac-address-increment = <1>;` to &eth0 to fix the MAC addresses, nothing else.